api/apis.py

```diff
@@ -0,0 +1,97 @@
+from re import search
+from rest_framework import viewsets, filters
+from rest_framework.permissions import IsAuthenticated
+from .models import Merchant, Store, Item, Category
+from .serializers import MerchantSerializer, StoreSerializer, ItemSerializer, CategorySerializer
+from django_filters.rest_framework import DjangoFilterBackend
+
+
+class MerchantViewset(viewsets.ModelViewSet):
+    queryset = Merchant.objects.all()
+    serializer_class = MerchantSerializer
+    permission_classes = [IsAuthenticated]
+    filter_backends = (filters.SearchFilter,)
+    search_fields = ['name']
+
+    # def get_queryset(self):
+    #     queryset = self.queryset
+    #     # if self.request.user.is_authenticated:
+    #     #     queryset = queryset.filter(user=self.request.user)
+    #     return queryset
+
+    def perform_create(self, serializer):
+        serializer.save(created_by=self.request.user)
+
+    def perform_update(self, serializer):
+        serializer.save(updated_by=self.request.user)
+        return super().perform_update(serializer)
+
+    # def get_permissions(self):
+    #     if self.action == 'list':
+    #         permission_classes = [IsAuthenticated]
+    #     else:
+    #         permission_classes = [IsAuthenticated]
+    #     return [permission() for permission in permission_classes]
+
+
+class StoreViewset(viewsets.ModelViewSet):
+    queryset = Store.objects.all()
+    serializer_class = StoreSerializer
+    permission_classes = [IsAuthenticated]
+    filter_backends = (DjangoFilterBackend, filters.SearchFilter)
+    filterset_fields = ['merchant', 'name']
+    search_fields = ['name', 'merchant__name']
+
+    def perform_create(self, serializer):
+        serializer.save(created_by=self.request.user)
+
+    def perform_update(self, serializer):
+        serializer.save(updated_by=self.request.user)
+        return super().perform_update(serializer)
```

Comment:
Case:
User1 creator of Merchant1, so is User2 for Merchant2.
Now, this API allows User2 create StoreX with Merchant1 info.